src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +39,328 @@ const ActorInit En_Door_InitVars = {
+        }
+        // Replace the door type it was loaded with by the new type
+        this->actor.params = (this->actor.params & ~0x380) | (doorType << 7);
+    }
+}
+
+#ifdef NON_MATCHING
+// Regalloc only
+void EnDoor_Idle(EnDoor* this, GlobalContext *globalCtx) {
+    Player* player = PLAYER;
+    s32 doorType;
+    Vec3f sp2C;
+    s8 numKeys;
+    s16 phi_v0;
+    s16 temp_v0;
+
+    doorType = (this->actor.params >> 7) & 7;
+    func_8002DBD0(&this->actor, &sp2C, &player->actor.posRot.pos);
+    if (this->unk_191 != 0) {
+        this->actionFunc = EnDoor_Open;
+        SkelAnime_ChangeAnimPlaybackStop(&this->skelAnime, D_809FCECC[this->unk_190], (((s32)player->stateFlags1 * 0x10) < 0) ? 0.75f : 1.5f);
+        if (this->lockTimer != 0) {
+            gSaveContext.dungeonKeys[gSaveContext.mapIndex]--;
+            Flags_SetSwitch(globalCtx, this->actor.params & 0x3F);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHAIN_KEY_UNLOCK);
+        }
+    } else if (!Player_InCsMode(globalCtx)) {
+        if (fabsf(sp2C.y) < 20.0f && fabsf(sp2C.x) < 20.0f && fabsf(sp2C.z) < 50.0f) {
+            phi_v0 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+            if (sp2C.z > 0.0f) {
+                phi_v0 = 0x8000 - phi_v0;
+            }
+            if (ABS(phi_v0) < 0x3000) {
+                if (this->lockTimer != 0) {
+                    numKeys = gSaveContext.dungeonKeys[gSaveContext.mapIndex];
+                    if (numKeys <= 0) {
+                        PLAYER->naviMessageId = -0x203;
+                        return;
+                    } else {
+                        player->doorTimer = 10;
+                    }
+                }
+                player->doorType = (doorType == DOOR_AJAR) ? -1 : 1;
+                player->doorDirection = (sp2C.z >= 0.0f) ? 1.0f : -1.0f;
+                player->doorActor = &this->actor;
+            }
+        } else if (doorType == DOOR_AJAR && this->actor.xzDistFromLink > 2*DOOR_SLAM_RANGE) {
+            // Once the player moves sufficiently far, have the door open slightly again
+            this->actionFunc = EnDoor_IdleAjar;
+        }
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Idle.s")
+#endif
+
+void EnDoor_WaitForCheck(EnDoor* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) { // if actor is checked
+        this->actionFunc = EnDoor_Talk;
+    } else {
+        // allows the player to "check" the door if within 40 units
+        func_8002F2CC(&this->actor, globalCtx, DOOR_CHECK_RANGE);
+    }
+}
+
+void EnDoor_Talk(EnDoor* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx)) {
+        // Return to waiting to be checked once text is dismissed
+        this->actionFunc = EnDoor_WaitForCheck;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Destroy.s")
+void EnDoor_CheckRange(EnDoor* this, GlobalContext* globalCtx) {
```

Comment:
the CheckRange name is kind of confusing with the unrelated "Check" from eg `EnDoor_WaitForCheck`?

I think action functions should be named to describe the state they're in, not what they're going to do next
for this one I'd suggest `EnDoor_AjarPlayerFar` or `EnDoor_AjarOpen`? (the latter being consistent with following suggestions)